### PR TITLE
fix: warning when package name unmatch from git

### LINKED ISSUE #237: npminstall don't work at mocha
```bash
$ git clone mochajs/mocha
$ npminstall
```

image

### PR DESCRIPTION
**Pull request by dead-horse**
closes #237 

This change is [Reviewable](https://reviewable.io/reviews/cnpm/npminstall/238)

### CODE REVIEW

File: lib/download/git.js [hunk 1]
```diff
@@ -41,7 +41,8 @@ module.exports = function* (pkg, options) {
     });
     const res = yield utils.copyInstall(cloneDir, options);
     if (pkg.name && pkg.name !== res.package.name) {
-      throw new Error(`Invalid Package, expected ${pkg.name} but found ${res.package.name}`);
+      options.console.warn(chalk.yellow(`[${pkg.displayName}] Package name unmatched: expected ${pkg.name} but found ${res.package.name}`));
```
Comment (atian25) [inline lib/download/git.js]: `will not install` ?

File: test/installGit.test.js [hunk 2]
```diff
@@ -146,16 +148,19 @@ describe('test/installGit.test.js', function() {
...
-      assert(/Invalid Package, expected error but found nunjucks/.test(err.message), err.message);
-    }
+
+  it('should warn on some name not match', done => {
+    coffee.fork(installBin, [
+      'error@git+https://github.com/mozilla/nunjucks.git#0f8b21b8d',
+    ], {
+      cwd: tmp,
+    })
+    .debug()
```
Comment (atian25) [inline test/installGit.test.js]: remove `debug`

File: lib/download/git.js (hunk 1 above)
Comment (dead-horse) [inline lib/download/git.js]: 会安装

Comment (dead-horse) [inline lib/download/git.js]: 只是提示名字不对

**Comment by fengmk2** [comment]
还需要修复到 https://github.com/cnpm/npminstall/tree/2.x

@dead-horse 你先 rebase master

**Pull request merged**